samza-rest/src/main/java/org/apache/samza/monitor/JobsClient.java

```diff
@@ -79,7 +80,7 @@ public JobsClient(List<String> jobStatusServers) {
    */
   public JobStatus getJobStatus(JobInstance jobInstance) {
     Job job = retriableHttpGet(baseUrl -> String.format(ResourceConstants.GET_JOBS_URL, baseUrl,
-        jobInstance.getJobName(), jobInstance.getJobId()));
+        jobInstance.getJobName(), jobInstance.getJobId()), new TypeReference<Job>(){});
```

Comment:
Why do you need a TypeReference? Why does retriableHttpGet need to be generic? It's a private method, used once, and its logic and logging makes it specific to fetching tasks.